### ISSUE TITLE
minishift start timesout on downloading ISO

### ISSUE THREAD

**Issue by danseo-reisys** [minishift/minishift#3005]
### General information

  * Minishift version: 1.27.1
  * OS: Windows
  * Hypervisor: VirtualBox

### Steps to reproduce
Hi,
Question1 - minishift start times out on downloading ISOs.   it starts and hangs in the middle and I need to reissue the start command but it hangs again.   it repeats and stops on different MB completion.  is it because GitHub is overloaded?  our company internet connection has no issue.    
it worked on one windows machine so far.   I have two additional windows machine to load minishift but failing.

Question2. - there is no document on this but do we need to turn off windows firewalls before we run minishift start?  

Question3 - also, do we need to add ssh client feature on windows before starting minishift start?  the pre-req steps for window has no mention on this.  

thanks

**Comment by danseo-reisys** [comment]
can we predownload the ISOs manually and flag minishift start command?

**Comment by gbraad** [comment]
> minishift start times out on downloading ISOs. 

Local network issues. It is unlikely that Github is the issue as they are providing you an endpoint from one fot he 3 datacenters based on geolocation. It is likely related to the company network. You can however copy the image and place it in the cache folder of the other machines.

> do we need to turn off windows firewalls before we run minishift start?

This really depends on how it is configured. In general the firewall will allow local connectivity and allows use of 192.168.x.x. But certain tools can affect the configuration as they hook into the network stack, like Symantec Endpoint Protection or McAfee...

> do we need to add ssh client feature on windows before starting minishift start? 

this is not a necessity as `minishift` has an internal ssh client. However, it is possible to use an external client... although on Windows this is often hit or miss as there are many different compiled versions scattered all over the Interweb. We have tested with cygwin, openssh-win32 (as originaly provided for powershell) and the beta of the ssh from optional features in win10. There were some others, but this is what works. Using just a random version from somewhere can become messy and even fail deployments.

**Comment by gbraad** [comment]
> can we predownload the ISOs manually and flag minishift start command?

yes. in your user profile there is a `.minishift/cache/` folder. this contains the ISO and `oc` binary used. you can also alternatively use `--iso-url file://c:/` to locate the file on disk.

**Comment by danseo-reisys** [comment]
can you give me complete example of --iso-url file://xxxx

**Comment by gbraad** [comment]
https://docs.okd.io/latest/minishift/using/basic-usage.html

**Comment by gbraad** [comment]
Does this answer you question? I would like to close this issue, as it is a duplicate of other reports when people were downloading from a bad/unreliable internet connection; downloads are as reliable and as fast as the lowest common denominator. The worst link determines how reliable it is... and there is not much we can do besides the offered caching. If you have a related question, please add...
